api/src/utils/getDAppNodeDomain.ts

```diff
@@ -13,12 +13,15 @@ export async function getDAppNodeDomain(): Promise<string> {
     return fs.readFileSync(path, 'utf-8');
   }
   for (let i = 0; i < maxRetries; i++) {
-    const response = await axios.get(config.dappmanager_domain_url);
-    if (response.status === 200) {
-      fs.writeFileSync(path, response.data, 'utf-8');
-      return response.data;
-    }
+    try {
+      const response = await axios.get(config.dappmanager_domain_url);
+      if (response.status === 200) {
+        fs.writeFileSync(path, response.data, 'utf-8');
+        return response.data;
+      }
+    } finally {
```

Comment:
In Javascript the finally clause is run before returning. So in this case it will wait 1s when finding the data which is not what we want. It won't catch the errors too

Reply:
What is your exact intention in this function's change?

Reply:
Without try block, if axios.get() throws an exception, there are no retries and function exits on the first (re)try.

Reply:
Okay then you want a try / catch block, like
```ts
for () {
  try {
    return await mayThrow()
  } catch (e) {
    // maybe handle error
  }
}
```

Reply:
So just shall we do it with empty catch block? Because we expect that that call could fail, but there is nothing else to do except to wait and try again in second.